packages/@react-spectrum/list/stories/ListView.stories.tsx

```diff
@@ -107,4 +112,121 @@ storiesOf('ListView', module)
+      <Item textValue="row1">row 1</Item>
+      <Item textValue="row2">row 2</Item>
+      <Item textValue="row3">row 3</Item>
+    </ListView>
+  ))
+  .add('density: spacious', () => (
+    <ListView width="250px" density="spacious">
+      <Item textValue="row1">row 1</Item>
+      <Item textValue="row2">row 2</Item>
+      <Item textValue="row3">row 3</Item>
+    </ListView>
+  ))
+  .add('selection: none', () => (
+    <Example selectionMode="none" />
+  ))
+  .add('selection: single, checkbox', () => (
+    <Example selectionMode="single" />
+  ))
+  .add('selection: single, checkbox, disabled', () => (
+    <Example selectionMode="single" disabledKeys={['row1']} />
+  ))
+  .add('selection: multiple, checkbox', () => (
+    <Example selectionMode="multiple" />
+  ))
+  .add('parent link example', () => (
+    <Example2 selectionMode="multiple" />
+  ))
+  .add('actions: ActionButton', () =>
+    renderActionsExample(props => <ActionButton {...props}><Copy /></ActionButton>))
+  .add('actions: ActionGroup', () =>
+    renderActionsExample(props => (
+      <ActionGroup {...props}>
+        <Item key="add"><Add /></Item>
+        <Item key="delete"><Delete /></Item>
+      </ActionGroup>
+    )))
+  .add('actions: ActionMenu', () =>
+    renderActionsExample(props => (
+      <ActionMenu {...props}>
+        <Item key="add">
+          <Add />
+          <Text>Add</Text>
+        </Item>
+        <Item key="delete">
+          <Delete />
+          <Text>Delete</Text>
+        </Item>
+      </ActionMenu>
+    )));
+
+function Example(props?) {
+  return (
+    <ListView width="250px" onAction={action('onAction')} {...props}>
+      <Item key="folder1" hasChildItems>
+        <Content>folder 1</Content>
+      </Item>
+      <Item key="row1">
+        <Content>row 1</Content>
+      </Item>
+      <Item key="row2">
+        <Content>row 2</Content>
+      </Item>
+      <Item key="row3">
+        <Content>row 3</Content>
+      </Item>
+    </ListView>
+  );
+}
+
+function Example2(props?) {
+  return (
+    <ListView width="250px" {...props}>
+      <Item key="folder1" hasChildItems>
+        <Link>folder 1</Link>
+      </Item>
+      <Item textValue="row1">
+        <Content>row 1</Content>
+      </Item>
+      <Item textValue="row2">
+        <Content>row 2</Content>
+      </Item>
+      <Item textValue="row3">
+        <Content>row 3</Content>
+      </Item>
+    </ListView>
+  );
+}
+
+function renderActionsExample(renderActions, props?) {
+  return (
+    <ListView width="300px" {...props}>
+      <Item textValue="row1" hasChildItems>
+        <Folder />
+        <Link>folder 1</Link>
+        <Text slot="description">description for folder 1</Text>
+        {renderActions({onPress: () => console.log('row 1')})}
+      </Item>
+      <Item textValue="row1">
+        <Text>row 1</Text>
+        <Text slot="description">description for row 1</Text>
+        {renderActions({onPress: () => console.log('row 1')})}
```

Comment:
Wonder if we need a way to make the actions in non-folder items line up with actions in folder items? Right now, they don't because folders have the chevron. I guess we'd either always reserve space for the chevron, or somehow detect if any of the items in the list are folders and only do it then. But that might be weird if you scrolled down and loaded more items and everything shifted when a folder came in.

Reply:
is it unreasonable to let it behave the way you say here (which is how subgrids would behave in the same circumstance) and they'd auto adjust when we detect a folder. Then have a prop that says I may have some folders where we'd reserve the space ahead of time so we don't get the shifting?

Reply:
Chatted about this with Alan, and he mentioned that if an item was a parent item, then the action buttons wouldn't line up. So, this is expected?